pennylane/ops/qubit/hamiltonian.py

```diff
@@ -166,6 +166,14 @@ class Hamiltonian(Observable):
     batch_size = None
     ndim_params = None  # could be (0,) * len(coeffs), but it is not needed. Define at class-level
 
+    def _flatten(self):
+        # note that we are unable to restore grouping type or method without creating new properties
```

Comment:
is saving the `_grouping_indices` not enough? then in `_unflatten` you can say:
```
op = cls(data[0], data[1])
op._grouping_indices = metadata[0]
```

Reply:
grouping indices is now metadata.  I needed to convert it all to tuples of tuples to get it to work, but should be fine now.